test/algorithms/simulation/sidb/operational_domain.cpp

```diff
@@ -85,22 +85,103 @@ TEST_CASE("BDL wire operational domain computation", "[operational-domain]")
+        }
+        SECTION("flood_fill")
+        {
+            const auto op_domain = operational_domain_flood_fill(lat, std::vector<tt>{create_id_tt()}, 1,
+                                                                 op_domain_params, &op_domain_stats);
+
+            // check if the operational domain has the correct size
+            CHECK(op_domain.operational_values.size() == 1);
+
+            // for the selected range, all samples should be within the parameters and operational
+            check_op_domain_params_and_operational_status(op_domain, op_domain_params, operational_status::OPERATIONAL);
+
+            CHECK(mockturtle::to_seconds(op_domain_stats.time_total) > 0.0);
+            CHECK(op_domain_stats.num_simulator_invocations == 2);
+            CHECK(op_domain_stats.num_evaluated_parameter_combinations == 1);
+            CHECK(op_domain_stats.num_operational_parameter_combinations == 1);
+            CHECK(op_domain_stats.num_non_operational_parameter_combinations == 0);
+        }
+        SECTION("contour_tracing")
+        {
+            const auto op_domain = operational_domain_contour_tracing(lat, std::vector<tt>{create_id_tt()}, 1,
+                                                                      op_domain_params, &op_domain_stats);
+
+            // check if the operational domain has the correct size
+            CHECK(op_domain.operational_values.size() == 1);
+
+            // for the selected range, all samples should be within the parameters and operational
+            check_op_domain_params_and_operational_status(op_domain, op_domain_params, operational_status::OPERATIONAL);
+
+            CHECK(mockturtle::to_seconds(op_domain_stats.time_total) > 0.0);
+            CHECK(op_domain_stats.num_simulator_invocations == 2);
+            CHECK(op_domain_stats.num_evaluated_parameter_combinations == 1);
+            CHECK(op_domain_stats.num_operational_parameter_combinations == 1);
+            CHECK(op_domain_stats.num_non_operational_parameter_combinations == 0);
+        }
+    }
+
     SECTION("operational area, same number of steps in x- and y-direction")
     {
         op_domain_params.x_min  = 5.1;
-        op_domain_params.x_max  = 6.1;
+        op_domain_params.x_max  = 6.0;
         op_domain_params.x_step = 0.1;
 
         op_domain_params.y_min  = 4.5;
-        op_domain_params.y_max  = 5.5;
+        op_domain_params.y_max  = 5.4;
         op_domain_params.y_step = 0.1;
 
         SECTION("grid_search")
         {
             const auto op_domain = operational_domain_grid_search(lat, std::vector<tt>{create_id_tt()},
                                                                   op_domain_params, &op_domain_stats);
 
-            // check if the operational domain has the correct size (10 steps in each dimension)
+            // check if the operational domain has the correct size
             CHECK(op_domain.operational_values.size() == 100);
```

Comment:
@marcelwa I think this did not make sense because with the limits set we would expect to have 121 values. This was caused by saying ``x_indices(num_x_steps())`` in the constructor. However, I think the non-deterministic behavior occurs because there is no check above to see if ``!current_neighborhood.empty()``. I hope this makes the changes in this PR a bit clearer.